unittests/runtime/Metadata.cpp

```diff
@@ -297,12 +297,12 @@ ProtocolDescriptor ProtocolNoWitnessTable{
 };
 
 TEST(MetadataTest, getExistentialMetadata) {
-  const ProtocolDescriptor *protoList1[] = {};
+  const ProtocolDescriptor *const proto = nullptr;
   RaceTest_ExpectEqual<const ExistentialTypeMetadata *>(
     [&]() -> const ExistentialTypeMetadata * {
       auto any = swift_getExistentialTypeMetadata(ProtocolClassConstraint::Any,
                                                   /*superclass=*/nullptr,
-                                                  0, protoList1);
+                                                  0, &proto);
```

Comment:
Why this change?

Reply:
Because the following is generated by the default warnings with clang/LLVM top-of-tree:
```
unittests/runtime/Metadata.cpp:300:44: warning: zero size arrays are an extension [-Wzero-length-array]
  const ProtocolDescriptor *protoList1[] = {};
                                           ^
```

Reply:
I briefly looked into that, but I lost interest in verifying that all of the downstream APIs could handle the array itself being null. In theory, none of this matters. Everything should be checking the "num protos" parameter first.

Reply:
That's exactly why I'd pass null here. If it doesn't cause any issues, it's a better test, and if it *does* cause issues you found a bug (which you aren't responsible for fixing).

Reply:
Well, 10975 tests passed on my Linux box, so let's go with `nullptr`. :-)